futurex_openedx_extensions/helpers/export_data.py

```diff
@@ -0,0 +1,179 @@
+
+
+def _get_mocked_request(url: str, user: Any, fx_info: dict, query_params: dict) -> Request:
+    """Create mocked request"""
+    factory = APIRequestFactory()
+    mocked_request = factory.get(url)
+    mocked_request.user = user
+    mocked_request.fx_permission_info = fx_info
+    mocked_request.query_params = query_params
+    return mocked_request
+
+
+def _get_user(user_id: int) -> Any:
+    """Get User from user_id"""
+    if user_id:
+        try:
+            return User.objects.get(id=user_id)
+        except User.DoesNotExist:
+            # Error handling need to be done
+            pass
+    return None
+
+
+def _get_view_class_instance(view_name: str) -> Any:
+    """Create view class instance from given iew name"""
+    view_instance = None
+    dasboard_views_module = importlib.import_module('futurex_openedx_extensions.dashboard.views')
+    view_class = getattr(dasboard_views_module, view_name, None)
+    if view_class:
+        view_instance = view_class.as_view()
+    return view_instance
+
+
+def export_data_to_csv(url: str, view_data: dict, fx_permission_info: dict, filename: str) -> tuple:
+    """
+    Mock view with given view params and write JSON response to CSV
+
+    :param url: view url
+    :param view_data: required data for mocking
+    :param fx_permission_info: contains role and permission info
+    :param fx_permission_info: filename for generated CSV
+
+    :return: generated filename
+    """
+    user = _get_user(view_data.get('user_id', None))
+    if not user:
+        return ('', f'Unable to get user for user_id: {view_data.get("user_id")}')
+
+    view_instance = _get_view_class_instance(view_data.get('view_class_name', ''))
+    if not view_instance:
+        return (
+            '',
+            f'Unable to create view class with given view name: {view_data.get("view_class_name")}'
+        )
+
+    query_params = view_data.get('query_params', {})
+    kwargs = view_data.get('kwargs', {})
+    # Ensure the filename ends with .csv
+    if not filename.endswith('.csv'):
+        filename += '.csv'
+
+    csv_file_path = os.path.join(settings.MEDIA_ROOT, filename)
+    with open(csv_file_path, mode='w', newline='', encoding='utf-8') as file:
+        # process first page and get header from the first page
+        mocked_request = _get_mocked_request(url, user, fx_permission_info, query_params)
+        response = view_instance(mocked_request, **kwargs)
+        if not response.status_code == 200:
+            return ('', f'View returned status code: {response.status_code}')
+        if not response.data or not response.data.get('results'):
+            return ('', 'View returned empty result')
+        header = response.data.get('results')[0].keys()
+        writer = csv.DictWriter(file, fieldnames=header)
+        writer.writeheader()
+        writer.writerows(response.data.get('results'))
+        # process next pages
+        while response.data.get('next'):
```

Comment:
please note that this logic assumes we always have pagination. This is the case for the two important views in our hands. However, the function should handle the case where no pagination class is associated with the view. Even if it just raises a `NotImplemented` exception